WalletWasabi.Tests/RegTests.cs

```diff
@@ -1219,71 +1246,60 @@ public async Task BuildTransactionValidationsTestAsync()
 
 			// toSend cannot have a zero element
-			Assert.Throws<ArgumentException>(() => wallet.BuildTransaction(null, new WalletService.Operation[0], 0));
+			Assert.Throws<ArgumentException>(() => wallet.BuildTransaction(null, new PaymentIntent(new DestinationRequest[] { }), FeeStrategy.SevenDaysConfirmationTargetStrategy));
 
 			// feeTarget has to be in the range 0 to 1008
-			Assert.Throws<ArgumentOutOfRangeException>(() => wallet.BuildTransaction(null, validOperationList, -10));
-			Assert.Throws<ArgumentOutOfRangeException>(() => wallet.BuildTransaction(null, validOperationList, 2000));
-
-			// subtractFeeFromAmountIndex has to be valid
-			Assert.Throws<ArgumentOutOfRangeException>(() => wallet.BuildTransaction(null, validOperationList, 2, null, false, -10));
-			Assert.Throws<ArgumentOutOfRangeException>(() => wallet.BuildTransaction(null, validOperationList, 2, null, false, 1));
+			Assert.Throws<ArgumentOutOfRangeException>(() => wallet.BuildTransaction(null, validIntent, FeeStrategy.CreateFromConfirmationTarget(-10)));
+			Assert.Throws<ArgumentOutOfRangeException>(() => wallet.BuildTransaction(null, validIntent, FeeStrategy.CreateFromConfirmationTarget(2000)));
 
 			// toSend amount sum has to be in range 0 to 2099999997690000
-			Assert.Throws<ArgumentOutOfRangeException>(() => wallet.BuildTransaction(null, invalidOperationList, 2));
+			Assert.Throws<ArgumentOutOfRangeException>(() => wallet.BuildTransaction(null, invalidIntent, FeeStrategy.TwoHoursConfirmationTargetStrategy));
 
 			// toSend negative sum amount
-			var operations = new[] { new WalletService.Operation(scp, Money.Satoshis(-10000), "") };
-			Assert.Throws<ArgumentException>(() => wallet.BuildTransaction(null, operations, 2));
+			Assert.Throws<ArgumentOutOfRangeException>(() => wallet.BuildTransaction(null, new PaymentIntent(scp, Money.Satoshis(-10000)), FeeStrategy.TwoHoursConfirmationTargetStrategy));
 
 			// toSend negative operation amount
-			operations = new[]
-			{
-				new WalletService.Operation(scp, Money.Satoshis(20000), ""),
-				new WalletService.Operation(scp, Money.Satoshis(-10000), "")
-			};
-			Assert.Throws<ArgumentException>(() => wallet.BuildTransaction(null, operations, 2));
+			Assert.Throws<ArgumentOutOfRangeException>(() => wallet.BuildTransaction(
+				null,
+				new PaymentIntent(
+					new DestinationRequest(scp, Money.Satoshis(20000)),
+					new DestinationRequest(scp, Money.Satoshis(-10000))),
+				FeeStrategy.TwoHoursConfirmationTargetStrategy));
 
 			// allowedInputs cannot be empty
-			Assert.Throws<ArgumentException>(() => wallet.BuildTransaction(null, validOperationList, 2, null, false, null, null, new TxoRef[0]));
+			Assert.Throws<ArgumentException>(() => wallet.BuildTransaction(null, validIntent, FeeStrategy.CreateFromConfirmationTarget(2), allowedInputs: new TxoRef[0]));
```

Comment:
Why we didn't use `FeeStrategy.TwoHoursConfirmationTargetStrategy`?

Reply:
Because I added that shortcut after I wrote down this line:) Fixed!